### PR TITLE
Implement penalize_validator helper methods for block processing.

### PR DESCRIPTION
**Pull request by schroedingerscode**
## PR Description
Implement penalize_validator helper method for block processing.

## Fixed Issue(s)
Not sure there's one specific issue for this PR, but it contributes to getting block processing updated to spec v0.1.

### CODE REVIEW

File: ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java [hunk 1]
```diff
@@ -292,6 +292,42 @@ void processDepositTopsUpValidatorBalanceWhenPubkeyIsFoundInRegistry() {
         beaconState.getValidator_balances().get(originalValidatorBalancesSize - 1));
   }
 
+  @Test
+  @Disabled // Pending resolution of Issue #347.
+  void penalizeValidatorDecrementsBackActorAndIncrementsWhistleblower() {
```
Comment (akhila-raju) [inline ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java]: Should this be called 'Bad' not 'Back'?

File: ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java [hunk 2]
```diff
@@ -292,6 +292,42 @@ void processDepositTopsUpValidatorBalanceWhenPubkeyIsFoundInRegistry() {
...
+        BeaconStateUtil.get_beacon_proposer_index(beaconState, beaconState.getSlot());
+    UnsignedLong whistleblowerReward =
+        BeaconStateUtil.get_effective_balance(beaconState, validatorIndex)
+            .dividedBy(UnsignedLong.valueOf(Constants.WHISTLEBLOWER_REWARD_QUOTIENT));
+    UnsignedLong whistleblowerBalance = beaconState.getValidator_balances().get(whistleblowerIndex);
+
+    UnsignedLong validatorBalance = beaconState.getValidator_balances().get(validatorIndex);
+
+    UnsignedLong expectedWhistleblowerBalance = whistleblowerBalance.plus(whistleblowerReward);
+    UnsignedLong expectedValidatorBalance = validatorBalance.minus(whistleblowerReward);
```
Comment (akhila-raju) [inline ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java]: Would you mind calling this expectedBadActorBalance? So it's consistent with the test name as well - less confusing 🙂

**Review by akhila-raju — CHANGES_REQUESTED** [review]
Just a few comments - otherwise LGTM

File: ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java (hunk 1 above)
Comment (schroedingerscode) [reply ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java]: 🤦‍♂️ Yes. Thanks.

File: ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java (hunk 2 above)
Comment (schroedingerscode) [reply ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/util/BeaconStateUtilTest.java]: 👍

**Comment by schroedingerscode** [comment]
Thanks, issues you've commented on have been fixed.

**Review by akhila-raju — APPROVED** [review]
LGTM

**Pull request merged**